lib/cancan/controller_resource_finder.rb

```diff
@@ -13,7 +13,9 @@ def find_resource
     end
 
     def find_resource_using_find_by
-      if resource_base.respond_to? 'find_by'
+      if resource_base.respond_to? "find_by_#{@options[:find_by]}!"
+        resource_base.send("find_by_#{@options[:find_by]}!", id_param)
+      elsif resource_base.respond_to? 'find_by'
```

Comment:
Revived it from here https://github.com/CanCanCommunity/cancancan/commit/33df7eb51a85692e702dc0e2a7252471bfe271ef#diff-38607cbe9e3608bd14d9367c9cf91eefL110